### PR TITLE
Add express routes

### PR DESCRIPTION
**Pull request by aggarwalpulkit596**

### CODE REVIEW

File: backend/entity/user.ts [hunk 1]
```diff
@@ -45,4 +45,7 @@ export class User {
 
   @UpdateDateColumn()
   updatedAt: Date;
+
+  @Column({ default: 0 })
+  totalPoints: number;
```
Comment (aggarwalpulkit596) [inline backend/entity/user.ts:50]: This was added here to show the leaderboard without complex calculations.

File: backend/repositories/submission.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { Submission } from '../entity';
+import { EntityRepository, Repository } from 'typeorm';
+@EntityRepository(Submission)
+class SubmissionRepository extends Repository<Submission> {
+  async saveWithAssets(req, payload): Promise<Submission> {
+    return this.save({
+      ...payload,
+      submittedAt: '2021-01-27T05:16:23.000Z',
```
Comment (aggarwalpulkit596) [inline backend/repositories/submission.ts]: How to get the time here ?

File: backend/repositories/submission.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { Submission } from '../entity';
+import { EntityRepository, Repository } from 'typeorm';
+@EntityRepository(Submission)
+class SubmissionRepository extends Repository<Submission> {
+  async saveWithAssets(req, payload): Promise<Submission> {
```
Comment (jatinkatyal13) [inline backend/repositories/submission.ts]: why have request object in repository?

and can we have type for payload as well?

File: backend/entity/submission.ts [hunk 4]
```diff
@@ -15,13 +8,11 @@ export class Submission {
...
-  })
-  status: SubmissionStatus;
-
   @Column({ type: 'timestamp' })
   submittedAt: Date;
+
+  @OneToMany(() => SubmissionAsset, (submissionAsset) => submissionAsset.submission, {
+    cascade: true,
+  })
+  submissionAsset: SubmissionAsset[];
```
Comment (jatinkatyal13) [inline backend/entity/submission.ts:17]: nitpick but should be plural

File: backend/repositories/user-task.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  async findById(id: number): Promise<UserTask> {
+    const userTask = await this.findOne(id, {
+      relations: ['task', 'submission'],
+    });
+    if (!userTask) throw new Error('ERR_USER_TASK_NOT_FOUND');
+    return userTask;
+  }
+
+  async findAll(): Promise<UserTask[]> {
+    // TODO: handle pagination
```
Comment (jatinkatyal13) [inline backend/repositories/user-task.ts]: shouldn't pagination be controller's concern? find already provide us with limit and skip options. findAll and findById are kindof redundant here

File: backend/routes/api/user-task/controller.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
+import { Request, Response } from 'express';
+import { Repositories } from '../../../repositories/index';
+import { SubmissionStatus } from '../../../entity/user-task';
+
+class UserController {
+  async handleActiveTasks(req: Request, res: Response) {
+    res.json(
+      await Repositories.getInstance().userTask.find({
```
Comment (jatinkatyal13) [inline backend/routes/api/user-task/controller.ts]: add a serialization step, we don't need a base controller but surely need a serialization/sanitization api

**Review by jatinkatyal13 — CHANGES_REQUESTED** [review]

**Pull request merged**